### ISSUE TITLE
Got CUDA_ERROR_MISALIGNED_ADDRESS error

### ISSUE THREAD

**Issue by mariolew** [TensorVision/TensorVision#46]
Hi, I just got the error 
**E tensorflow/stream_executor/cuda/cuda_driver.cc:1140] could not synchronize on CUDA context: CUDA_ERROR_MISALIGNED_ADDRESS :: No stack trace available
F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:198] Unexpected Event status: 1**  
when running _tv-train --hypes examples/cifar10_vgg8.json_
It's probably a tensorflow issue, **_but I just can't reproduce this error by running any example of tensorflow. But when I tried to run a tensorvision example, I got an error like above_**.

I was using tensorflow built from source, and my GPU is Tesla K40c.

**Comment by MartinThoma** [comment]
@mariolew Did you try it more than once? Could you please paste the complete output, e.g.

``` bash
$ tv-train --hypes examples/cifar10_vgg8.json
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcuda.so locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcurand.so locally
2016-07-11 10:03:34,395 INFO No environment variable 'TV_PLUGIN_DIR' found. Set to '/home/moose/tv-plugins'.
2016-07-11 10:03:34,395 INFO No environment variable 'TV_STEP_SHOW' found. Set to '50'.
2016-07-11 10:03:34,395 INFO No environment variable 'TV_STEP_EVAL' found. Set to '250'.
2016-07-11 10:03:34,396 INFO No environment variable 'TV_STEP_WRITE' found. Set to '1000'.
2016-07-11 10:03:34,396 INFO No environment variable 'TV_MAX_KEEP' found. Set to '10'.
2016-07-11 10:03:34,396 INFO No environment variable 'TV_STEP_STR' found. Set to 'Step {step}/{total_steps}: loss = {loss_value:.2f} ( {sec_per_batch:.3f} sec (per Batch); {examples_per_sec:.1f} examples/sec)'.
2016-07-11 10:03:34,396 INFO f: <open file 'examples/cifar10_vgg8.json', mode 'r' at 0x7f648732a4b0>
2016-07-11 10:03:34,396 INFO Initialize training folder
2016-07-11 10:03:34,397 INFO Start training
Filling queue with 5000 CIFAR images before starting to train. This will take a few minutes.
I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:924] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
I tensorflow/core/common_runtime/gpu/gpu_init.cc:102] Found device 0 with properties: 
name: GeForce 940MX
major: 5 minor: 0 memoryClockRate (GHz) 1.2415
pciBusID 0000:02:00.0
Total memory: 2.00GiB
Free memory: 1.58GiB
I tensorflow/core/common_runtime/gpu/gpu_init.cc:126] DMA: 0 
I tensorflow/core/common_runtime/gpu/gpu_init.cc:136] 0:   Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:806] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce 940MX, pci bus id: 0000:02:00.0)
2016-07-11 10:03:40,384 INFO Step 0/100000: loss = 4.83 ( 0.082 sec (per Batch); 1566.4 examples/sec)
2016-07-11 10:03:49,444 INFO Step 50/100000: loss = 4.32 ( 0.169 sec (per Batch); 757.9 examples/sec)
2016-07-11 10:03:58,138 INFO Step 100/100000: loss = 4.11 ( 0.168 sec (per Batch); 761.4 examples/sec)
2016-07-11 10:04:06,499 INFO Step 150/100000: loss = 3.83 ( 0.162 sec (per Batch); 790.5 examples/sec)
2016-07-11 10:04:14,812 INFO Step 200/100000: loss = 3.72 ( 0.161 sec (per Batch); 794.5 examples/sec)
2016-07-11 10:04:22,861 INFO Doing Evaluate with Training Data.
2016-07-11 10:04:22,862 WARNING Passing eval_op directly is deprecated. Pass a list of tuples instead.
2016-07-11 10:04:22,862 INFO Data: train  Num examples:  49920
```

Which tensorflow version (`tensorflow.__version__`) do you have?

**Comment by mariolew** [comment]
Hi, @MartinThoma Thanks for you response.
The complete output is like below:

> TensorVision$ ./bin/tv-train --hypes=examples/cifar10_cnn.json 
> I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcublas.so.7.0 locally
> I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcudnn.so.4 locally
> I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcufft.so.7.0 locally
> I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcuda.so.1 locally
> I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcurand.so.7.0 locally
> 2016-07-11 16:08:36,907 INFO No environment variable 'TV_PLUGIN_DIR' found. Set to '/home/whdeng-k40/tv-plugins'.
> 2016-07-11 16:08:36,908 INFO No environment variable 'TV_STEP_SHOW' found. Set to '50'.
> 2016-07-11 16:08:36,908 INFO No environment variable 'TV_STEP_EVAL' found. Set to '250'.
> 2016-07-11 16:08:36,908 INFO No environment variable 'TV_STEP_WRITE' found. Set to '1000'.
> 2016-07-11 16:08:36,908 INFO No environment variable 'TV_MAX_KEEP' found. Set to '10'.
> 2016-07-11 16:08:36,908 INFO No environment variable 'TV_STEP_STR' found. Set to 'Step {step}/{total_steps}: loss = {loss_value:.2f} ( {sec_per_batch:.3f} sec (per Batch); {examples_per_sec:.1f} examples/sec)'.
> 2016-07-11 16:08:36,908 INFO f: <open file 'examples/cifar10_cnn.json', mode 'r' at 0x7f1bd0f706f0>
> 2016-07-11 16:08:36,908 INFO Initialize training folder
> 2016-07-11 16:08:36,909 INFO Start training
> Filling queue with 5000 CIFAR images before starting to train. This will take a few minutes.
> I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:925] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
> I tensorflow/core/common_runtime/gpu/gpu_init.cc:102] Found device 0 with properties: 
> name: Tesla K40c
> major: 3 minor: 5 memoryClockRate (GHz) 0.745
> pciBusID 0000:01:00.0
> Total memory: 11.25GiB
> Free memory: 11.09GiB
> I tensorflow/core/common_runtime/gpu/gpu_init.cc:126] DMA: 0 
> I tensorflow/core/common_runtime/gpu/gpu_init.cc:136] 0:   Y 
> I tensorflow/core/common_runtime/gpu/gpu_device.cc:839] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Tesla K40c, pci bus id: 0000:01:00.0)
> E tensorflow/stream_executor/cuda/cuda_event.cc:49] Error polling for event status: failed to query event: CUDA_ERROR_MISALIGNED_ADDRESS
> F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:198] Unexpected Event status: 1
> Aborted (core dumped)

And my tensorflow version is 0.9.0.

**Comment by MartinThoma** [comment]
At least it was noted on Tensorflow, too: https://github.com/tensorflow/tensorflow/issues/3224

`python -m tensorflow.models.image.mnist.convolutional` gives the same error for me. Strange. By now, I don't know what causes the problem / how to fix it.

**Comment by mariolew** [comment]
Yeah, so strange.
I've tracked this issue, it will raise the error when executing following lines:
 in train.py: line 329 sess_coll = core.start_tv_session(hypes)
in core.py: line 156 threads = tf.train.start_queue_runners(sess=sess, coord=coord)
